x-pack/plugins/security_solution/common/endpoint/service/authz/authz.ts

```diff
@@ -99,10 +100,19 @@ export const calculateEndpointAuthz = (
   const authz: EndpointAuthz = {
     canWriteSecuritySolution,
     canReadSecuritySolution,
+
+    // ---------------------------------------------------------
+    // Coming from Fleet authz
+    // ---------------------------------------------------------
     canAccessFleet: fleetAuthz?.fleet.all ?? false,
     canReadFleetAgentPolicies: fleetAuthz?.fleet.readAgentPolicies ?? false,
     canWriteFleetAgents: fleetAuthz?.fleet.allAgents ?? false,
     canReadFleetAgents: fleetAuthz?.fleet.readAgents ?? false,
+    canWriteIntegrationPolicies: fleetAuthz?.integrations.writeIntegrationPolicies ?? false,
```

Comment:
Just to make sure I understand this property: this to control whether a user can be re-directed to the Fleet Page to create an integration, correct?

Reply:
this, together with `canAccessFleet`, this is the condition to show the 'Add Elastic Defend` button:
```ts
{!authzLoading && canAccessFleet && canWriteIntegrationPolicies && (JSX comes here)}
```
here:
https://github.com/elastic/kibana/pull/199803/files#diff-7fd711c60fc2c52fc144c41961e49c3a2ecca8805c7ad065e3cbe9fa221dd730R141

because for creating an integration, integration:all is needed as well, although it is not needed afterwards to enroll an agent